vault/resource_kmip_secret_backend.go

```diff
@@ -0,0 +1,222 @@
+		Delete: kmipSecretBackendDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				Description:  "Path where KMIP secret backend will be mounted",
+				ValidateFunc: validateNoTrailingLeadingSlashes,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Human-friendly description of the mount for the backend",
+			},
+			"listen_addrs": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Description: "Addresses the KMIP server should listen on (host:port)",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"server_hostnames": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Hostnames to include in the server's TLS certificate as SAN DNS names. The first will be used as the common name (CN)",
+			},
+			"server_ips": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "IPs to include in the server's TLS certificate as SAN IP addresses",
+			},
+
+			"tls_ca_key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "CA key type, rsa or ec",
+			},
+			"tls_ca_key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "CA key bits, valid values depend on key type",
+			},
+			"tls_min_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Minimum TLS version to accept",
+			},
+			"default_tls_client_key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client certificate key type, rsa or ec",
+			},
+			"default_tls_client_key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client certificate key bits, valid values depend on key type",
+			},
+			"default_tls_client_ttl": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client certificate TTL in seconds",
+			},
+		},
+	}
+}
+
+func kmipSecretBackendCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Get("path").(string)
+	description := d.Get("description").(string)
+	defaultTLSClientTTL := d.Get("default_tls_client_ttl").(int)
+
+	log.Printf("[DEBUG] Mounting KMIP backend at %q", path)
+	if err := client.Sys().Mount(path, &api.MountInput{
+		Type:        "kmip",
+		Description: description,
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%d", defaultTLSClientTTL),
```

Comment:
Should we specify the units here `%ds` in duration notation?

Reply:
Probably could move the `fmt.Sprintf` out to where you assign `defaultTLSClientTTL` above.

Reply:
Fixed and added `%ds` notation in 3686e05